### PR TITLE
Mamba-2 model setup

### PR DESCRIPTION
**Pull request by MaizeHH**
- Setup mamba-2 model
- Validation data used for learning rate scheduler
- Config for easy use 
- Retrieval of data align with generation setup
- CI/CD & testing

### CODE REVIEW

**Review by TheMOHD3886 — APPROVED** [review]
lgtm 👍

**Review by MortenMunk — APPROVED** [review]
LGTM

File: src/train.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+            ciphertext = data["ciphertext"]
+            if isinstance(ciphertext, str):
+                ciphertext = [int(x) for x in ciphertext.split()]
+            plaintext = data["plaintext"]
+            encoded_plain = [self.mapping[c] + self.char_offset for c in plaintext.lower() if c in self.mapping]
+            cipher_tensor = torch.tensor(self._pad_trunc(ciphertext), dtype=torch.long)
+            plain_tensor = torch.tensor(self._pad_trunc(encoded_plain), dtype=torch.long)
+            return cipher_tensor, plain_tensor
+        except Exception:
+            return torch.zeros(self.max_seq_len, dtype=torch.long), torch.zeros(self.max_seq_len, dtype=torch.long)
```
Comment (Jatewo) [inline src/train.py]: I don't think this is the best approach. A malformed json will now cause 0's instead of crashing. I think we would rather have an error raised to catch malformed data

File: src/config.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    
+    batch_size: int = 128
+    learning_rate: float = 5e-4
+    epochs: int = 30
+
+    patience: int = 1
+    factor: float = 0.5
+
+    save_path: str = "./outputs"
+    data_dir: str = "../Ciphers/"
```
Comment (Jatewo) [inline src/config.py]: We should consider using e.g. pathlib for the paths to ensure proper handling. Now it depends on where you run the code from.

File: src/train.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        ciphertext = data.get("ciphertext", [])
+        if isinstance(ciphertext, str):
+            ciphertext = [int(x) for x in ciphertext.split()]
+        
+        actual_max_val = max(ciphertext) if ciphertext else 0
+        actual_len = len(ciphertext)
+        
+        return actual_len, actual_max_val
+    except Exception:
+        return 0, 0
```
Comment (Jatewo) [inline src/train.py]: This is pretty much the same issue as the torch.zeros

Comment (Jatewo) [reply src/train.py]: Perhaps log a warning that a file is skipped/corrupted

**Review by Jatewo — CHANGES_REQUESTED** [review]

**Review by Jatewo — APPROVED** [review]
I will approve but add an issue regarding refactoring of the very large functions

**Review by TheMOHD3886 — COMMENTED** [review]
lgtm 👍
